File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/shared/postauthorizeresponse.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class PostAuthorizeResponse:
    gateway_order_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('gatewayOrderId') }})
    result_code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('resultCode') }})
    result_message: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('resultMessage') }})
    success: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('success') }})
    transaction_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('transactionId') }})
    

File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/shared/postauthorizeresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class PostAuthorizeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=gatewayOrderId" })
  gatewayOrderId?: string;

  @SpeakeasyMetadata({ data: "json, name=resultCode" })
  resultCode?: string;

  @SpeakeasyMetadata({ data: "json, name=resultMessage" })
  resultMessage?: string;

  @SpeakeasyMetadata({ data: "json, name=success" })
  success?: boolean;

  @SpeakeasyMetadata({ data: "json, name=transactionId" })
  transactionId?: string;
}
